texar/utils/beam_search.py

```diff
@@ -36,8 +36,10 @@
 INF = 1.0 * 1e7
 
 
-def gather_nd(params: torch.Tensor, indices: torch.Tensor) -> torch.Tensor:
-
+def gather_nd(params: Union[torch.Tensor, int],
```

Comment:
Why would this and following methods take `int` as an argument?

Reply:
`state` used in `beam_search` is `AttentionWrapperState`. We here have `int` because of `time` in  `AttentionWrapperState`. I deleted `time` to simplify the interface in `beam_search`.

Reply:
I see. Yeah `time` is probably unnecessary, and I kind of think that some other fields in `AttentionWrapperState` are unnecessary but I'm not sure.

Reply:
`AttentionWrapperState` in TF [here](https://www.tensorflow.org/api_docs/python/tf/contrib/seq2seq/AttentionWrapperState) has `time`. (Is `time` used in TF?)  I'd suggest keeping it the same as the TF interface even if it's currently not used (it might be used in other attention mechanisms, probably in the future)

Reply:
If I remember correctly, in TF `time` is only used to keep track of the time step in the decoding procedure. If we really want to keep it the same as the TF interface, we could either make modifications in `beam_search` or `utils.map_structure.` (I probably can follow the similar way of `no_map_type` to bypass this issue).

Reply:
Yes, let's keep `time` as in TF. It's likely to be useful in the future.